### PR TITLE
Add scope document

### LINKED ISSUE #119: [FEATURE REQUEST]
As a user, I'd like to understand what's the vision and scope of AsyncAPI. I should be clear what is part of the scope and what i
[vision-document-template-v1.pdf](https://github.com/asyncapi/asyncapi/files/2835480/vision-document-template-v1.pdf)
s not, and where the project is heading to.

@jhigginbotham did some great questions that must be answered in this document:

* What is the overall scope?
* Is it just to capture events and streams available?
* Is it a higher-level view of all interaction styles, e.g. RPC, REST, message streams, events, file drops/FTP
* What is the end goal/stopping point for the spec - drive definition capture, doc generation, other?
* How does this fit with OAS?
* How does this fit with ALPS?
* Do we support any/all of the following technologies? e.g. gRPC (has bi-dir support), Websockets, SSE, Webhooks, Kafka, and Kinesis streams
* What formats do we support? e.g. JSON, XML, Avro, ProtoBuf, Binary

References:

* https://wiki.base22.com/btg/template-vision-and-scope-document-20480031.html
* http://trishmarie.com/vision.pdf
* See attached documents

### PR DESCRIPTION
**Pull request by fmvilas**
### What?
Fixes #119.

### Why?
[Let's make AsyncAPI a welcoming community](https://github.com/asyncapi/asyncapi/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22Make+AsyncAPI+a+welcoming+community%22)

### CODE REVIEW

File: SCOPE.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# AsyncAPI Scope
+
+AsyncAPI is a specification for defining message-driven APIs. We're on a mission to standardize message-based communication and increase interoperability of the different types of systems out there.
```
Comment (jhigginbotham) [inline SCOPE.md]: out there -> available

File: SCOPE.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Scope
+
+The following is a list of features the AsyncAPI specification and its official tooling should offer:
+
+- Provide enough information to connect and consume message-driven APIs. An AsyncAPI document should provide all your software needs to connect to a given API and start publishing and/or consuming messages from it.
```
Comment (jhigginbotham) [inline SCOPE.md]: "all your software needs" - as a specification, it isn't providing software. It is providing a description of the existing async APIs available for consumption, right?

File: SCOPE.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- Offer a framework for validating message payloads and headers.
+- Allow you to strictly define your message-driven API in a machine-readable format.
+- Provide mechanisms for mapping the information contained in an AsyncAPI document to the major messaging protocols.
```
Comment (jhigginbotham) [inline SCOPE.md]: mechanisms, or a definition language to describe protocols available for publishing and consuming messages?

File: SCOPE.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- Protocol-specific features, such as queues, exchanges, policies, partitions, etc; always via protocol mappers.
+- Provide a standard way to create specification extensions and promote their discoverability. 
+- Offer mechanisms to define message schemas in different formats. Namely: JSON Schema, OpenAPI schema (_slightly different than JSON Schema Draft 04_), Protobuf, Avro, etc.
+- Provide parsers for all the major programming languages, so it's easier for others to create tooling.
```
Comment (jhigginbotham) [inline SCOPE.md]: Perhaps separate the definition spec from the tooling, as not to create confusion like what we saw with Swagger vs. OAS?!

File: SCOPE.md (hunk 2 above)
Comment (fmvilas) [reply SCOPE.md]: Maybe like this?

An AsyncAPI document should provide **everything** your software needs to connect to a given API and start publishing and/or consuming messages from it.

It doesn't say it provides software. It provides all the information your software will need to connect and consume (maybe should be like this?). In any case, AsyncAPI also provides tooling for that purpose so it offers software, not just the spec.

File: SCOPE.md (hunk 3 above)
Comment (fmvilas) [reply SCOPE.md]: Both, a definition language and tools to process these definitions.

File: SCOPE.md (hunk 4 above)
Comment (fmvilas) [reply SCOPE.md]: I'd like to keep it part of the same thing. The problem between Swagger and OAS is that one is part of a company and the other part of a foundation. I want to keep both together, always. Most people don't care about specs, but they do about the tooling. Ultimately is what you use. That's one of the reasons it's being hard for the OAS folks to make people say OpenAPI and not Swagger.

File: SCOPE.md (hunk 3 above)
Comment (MikeRalphson) [reply SCOPE.md]: To clarify is "official tooling" both that which is originated within, or donated to, the AsyncAPI github organisation and no others?

Comment (fmvilas) [reply SCOPE.md]: So far yes. The ones the AsyncAPI is responsible for maintaining.

File: SCOPE.md (hunk 4 above)
Comment (jhigginbotham) [reply SCOPE.md]: Fair enough. May I suggest then to rework the scope and such and call out the spec + tooling as two separate offerings as part of the initiative? The bullets indicate this but aren't grouped, so it becomes confusing. I could see readers asking "is it a spec? Is it a solution? Is it tools?" Simply reorganizing and clarifying the intent will go far, I think.

Comment (fmvilas) [reply SCOPE.md]: Agree. It should be clear.

**Pull request closed without merging**